src/test/scala/ai/starlake/job/sink/pubsub/PubSubProviderTest.scala

```diff
@@ -0,0 +1,145 @@
+package ai.starlake.job.sink.pubsub
+
+import better.files.File
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.cloud.pubsub.v1.Publisher
+import com.google.cloud.pubsub.v1.stub.{GrpcSubscriberStub, SubscriberStubSettings}
+import com.google.protobuf.ByteString
+import com.google.pubsub.v1.{AcknowledgeRequest, ProjectSubscriptionName, ProjectTopicName, PubsubMessage, PullRequest}
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.HttpClientBuilder
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.streaming.MemoryStream
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.io.{FileInputStream, IOException}
+import scala.jdk.CollectionConverters.{IterableHasAsJava, ListHasAsScala}
+
+class PubSubProviderTest extends AnyFlatSpec with Matchers with StrictLogging {
+
+  // GCP authentication is made using the GOOGLE_APPLICATION_CREDENTIALS environment variable
+  private val projectId = "phenytech-global"
```

Comment:
Project id should not be hard coded but defaulted to the current project id
the subscription id should be created before this PR is merged